keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -546,6 +565,51 @@ class ClusterHandler(
           current!!.scaling.stepScalingPolicies != desired.scaling.stepScalingPolicies
         )
 
+  private fun ResourceDiff<ServerGroup>.disableOtherServerGroupJob(resource: Resource<ClusterSpec>, desiredVersion: String): Map<String, Any?> {
```

Comment:
This is the exact same logic as titus. Like, the exact same. This is what prompted me to think about how much of the cluster stuff can be common between cloud providers.

Reply:
You could always extract this into a common function in a shared module like `keel-core` (or even a new shared module between EC2 and Titus)... Also, FWIW, both those specs inherit from `ComputeResourceSpec`, so you could even think about extracting common relevant fields to that interface if it helps.

Reply:
Hmm good point. There might be some weirdness with typing for a common function (I'm not the best at generics). Let me think about this.

Reply:
Given that you need two types, and they're both abstract types on the resource handler, I don't think it's worth it to make a separate function. I'd rather think about how we can abstract some of the cluster logic into a generic cluster handler because I think that's where the functionality actually should live.